pkg/controller/statusmanager/machineconfig_watcher.go

```diff
@@ -0,0 +1,110 @@
+package statusmanager
+
+import (
+	"context"
+
+	"github.com/openshift/cluster-network-operator/pkg/platform"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+type MachineConfigsWatcher struct {
+	status *StatusManager
+}
+
+// AddMachineConfigsWatcher wires up the MachineConfigWatcher and MachineConfigPoolWatcher
+// to the controller-manager.
+func (s *StatusManager) AddMachineConfigsWatcher(mgr manager.Manager) error {
+	if s.hyperShiftConfig.Enabled {
+		// MachineConfig is not supported in HyperShift cluster, so return without
+		// initializing watcher.
+		return nil
+	}
+
+	pw := &MachineConfigsWatcher{
+		status: s,
+	}
+	c, err := controller.New("machineconfigs-watcher", mgr, controller.Options{Reconciler: pw})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(source.Kind[crclient.Object](mgr.GetCache(), &mcfgv1.MachineConfig{},
+		&handler.EnqueueRequestForObject{}, onMachineConfigPredicate()))
+	if err != nil {
+		return err
+	}
+
+	return c.Watch(source.Kind[crclient.Object](mgr.GetCache(), &mcfgv1.MachineConfigPool{},
+		&handler.EnqueueRequestForObject{}, onMachineConfigPoolPredicate()))
+}
+
+// Reconcile triggers a re-update of Status.
+func (p *MachineConfigsWatcher) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	defer utilruntime.HandleCrash(p.status.SetDegradedOnPanicAndCrash)
+	p.status.SetFromMachineConfigs(ctx)
+	return reconcile.Result{}, nil
+}
+
+func onMachineConfigPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			mc := e.Object.(*mcfgv1.MachineConfig)
+			return hasRequiredLabel(mc)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			mc := e.ObjectNew.(*mcfgv1.MachineConfig)
+			return hasRequiredLabel(mc)
+		},
+	}
+}
```

Comment:
Why is this predicate not based on the owner reference?
Why is this predicate not filtering delete events?